### PR TITLE
GCN event search: by partial dateobs and aliases

### PR DESCRIPTION
**Pull request by Theodlz**
This PR addresses #4300 for:
- candidate scanning
- gcnevents list page
- gcnevents crossmatch dialog

Also addresses #4336

### CODE REVIEW

File: skyportal/handlers/api/gcn.py [hunk 1]
```diff
@@ -1362,9 +1362,19 @@ async def get(self, dateobs=None):
...
+                    arrow.get(partialdateobs.strip()).datetime
+                    partialdateobs = partialdateobs.replace("T", " ")
+                except Exception:
+                    if len(partialdateobs) > 10 and partialdateobs[10] == "T":
+                        partialdateobs = partialdateobs.replace("T", " ")
+                        print(f"2. manage to parse {partialdateobs}")
+                partialdateobs = partialdateobs.strip().lower()
                 query = query.where(
                     cast(GcnEvent.dateobs, String).like(f"{partialdateobs}%")
+                    | func.lower(cast(GcnEvent.aliases, String)).like(
```
Comment (mcoughlin) [inline skyportal/handlers/api/gcn.py:1375]: This is neat.

File: skyportal/handlers/api/gcn.py [hunk 2]
```diff
@@ -1362,9 +1362,19 @@ async def get(self, dateobs=None):
...
 
             if partialdateobs is not None and partialdateobs != "":
-                partialdateobs = partialdateobs.replace("T", " ")
+                try:
+                    arrow.get(partialdateobs.strip()).datetime
+                    partialdateobs = partialdateobs.replace("T", " ")
+                except Exception:
+                    if len(partialdateobs) > 10 and partialdateobs[10] == "T":
+                        partialdateobs = partialdateobs.replace("T", " ")
+                        print(f"2. manage to parse {partialdateobs}")
```
Comment (mcoughlin) [inline skyportal/handlers/api/gcn.py:1371]: Can remove the print statement?

**Review by mcoughlin — APPROVED** [review]
Gonna be nice. Thanks in advance for making this the same in the crossmatch.

File: skyportal/handlers/api/gcn.py (hunk 2 above)
Comment (Theodlz) [reply skyportal/handlers/api/gcn.py:1371]: Oups. Will do

**Comment by Theodlz** [comment]
@mcoughlin i didnt see your bugfix PR (#4337), do fixed it here too + some backend and frontend improvements. 

Could you have a quicklook maybe?

**Pull request merged**